docs/source/conf.py

```diff
@@ -385,27 +387,41 @@ def generate_weights_table(module, table_name, metrics, dataset, include_pattern
     if exclude_patterns is not None:
         weights = [w for w in weights if all(p not in str(w) for p in exclude_patterns)]
 
+    rich_metadata = ["GFLOPs", "Size (MB)"]
+
     metrics_keys, metrics_names = zip(*metrics)
-    column_names = ["Weight"] + list(metrics_names) + ["Params", "Recipe"]
-    column_names = [f"**{name}**" for name in column_names]  # Add bold
+    column_names = ["Weight"] + list(metrics_names) + ["Params"] + rich_metadata + ["Recipe"]  # Final column order
+    column_names_table = [f"**{name}**" for name in column_names if name not in exclude_columns]  # Add bold
+
+    column_patterns = list(
+        zip(
+            column_names,
+            [lambda w: f":class:`{w} <{type(w).__name__}>`"]
+            + [lambda w, metric=m: w.meta["_metrics"][dataset][metric] for m in metrics_keys]
+            + [
+                lambda w: f"{w.meta['num_params']/1e6:.1f}M",
+                lambda w: f"{w.meta['_flops']:.3f}",
+                lambda w: f"{round(w.meta['_weight_size'], 1):.1f}",
+                lambda w: f"`link <{w.meta['recipe']}>`__",
+            ],
+        )
+    )
 
     content = [
-        (
-            f":class:`{w} <{type(w).__name__}>`",
-            *(w.meta["_metrics"][dataset][metric] for metric in metrics_keys),
-            f"{w.meta['num_params']/1e6:.1f}M",
-            f"`link <{w.meta['recipe']}>`__",
-        )
-        for w in weights
+        [pattern(w) for col_name, pattern in column_patterns if col_name not in exclude_columns] for w in weights
     ]
-    table = tabulate(content, headers=column_names, tablefmt="rst")
+
+    column_widths = zip(column_names, ["120"] + ["18"] * len(metrics_names) + ["18", "18", "18", "10"])
```

Comment:
Why are number of `18`s is fixed? Shouldn't it depend on whether `rich_metadata` are included for this model?

Reply:
This generates widths for all existing columns, which are then filtered in the line below and saved to `widths_table`.

Reply:
Gotcha, you are right.